src/test/util/txmempool.cpp

```diff
@@ -68,10 +69,24 @@ std::optional<std::string> CheckPackageMempoolAcceptResult(const Package& txns,
             return strprintf("tx %s unexpectedly failed: %s", wtxid.ToString(), atmp_result.m_state.ToString());
         }
 
-        //m_replaced_transactions should exist iff the result was VALID
-        if (atmp_result.m_replaced_transactions.has_value() != valid) {
-            return strprintf("tx %s result should %shave m_replaced_transactions",
+        if (atmp_result.m_replaced_transactions.has_value()) {
+            // m_replaced_transactions should exist iff the result was VALID
+            if (!valid) {
+                return strprintf("tx %s result should %shave m_replaced_transactions",
                                     wtxid.ToString(), valid ? "" : "not ");
+            }
+            // Each subpackage is allowed MAX_REPLACEMENT_CANDIDATES replacements (only checking individually here)
+            if (atmp_result.m_replaced_transactions.value().size() > MAX_REPLACEMENT_CANDIDATES) {
+                return strprintf("tx %s result replaced too many transactions",
+                                    wtxid.ToString());
+            }
+
+            // Replacements can't happen for subpackages larger than 2
+            if (!atmp_result.m_replaced_transactions.value().empty() &&
+                atmp_result.m_wtxids_fee_calculations.has_value() && atmp_result.m_wtxids_fee_calculations.value().size() > 2) {
+                 return strprintf("tx %s was part of a too-large package RBF subpackage",
```

Comment:
Maybe you can also check the topology/cluster size of this tx in mempool?

Reply:
added a small check here using `GetEntry` that takes `Wtxid`s.